### PR TITLE
Update Ballerina gradle plugin and centralized workflows

### LINKED ISSUE #5860: Update Ballerina gradle plugin and centralized workflows
**Description:**
$Subject

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
# Description

$Subject

Fixes https://github.com/ballerina-platform/ballerina-library/issues/5860

### CODE REVIEW

File: .github/workflows/build-with-bal-test-graalvm.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  cancel-in-progress: true
+
+jobs:
+  call_stdlib_workflow:
+    name: Run StdLib Workflow
+    if: ${{ github.event_name != 'schedule' || (github.event_name == 'schedule' && github.repository_owner == 'ballerina-platform') }}
+    uses: ballerina-platform/ballerina-standard-library/.github/workflows/build-with-bal-test-graalvm-connector-template.yml@main
+    secrets: inherit
+    with:
+      additional-build-flags: "-x :googleapis.gmail-examples:build"
```
Comment (NipunaRanasinghe) [inline .github/workflows/build-with-bal-test-graalvm.yml]: we seem to have gmail module references here. Please modify all the references to the excel module

File: ballerina/build.gradle [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+ * under the License.
+ */
+
+import org.apache.tools.ant.taskdefs.condition.Os
+
+plugins {
+    id 'io.ballerina.plugin'
+}
+
+description = 'Excel - Ballerina'
```
Comment (NipunaRanasinghe) [inline ballerina/build.gradle]:
```suggestion
description = 'Microsoft Excel - Ballerina'
```

File: gradle.properties [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+group=io.ballerina.stdlib
+version=3.0.0-SNAPSHOT
+
+checkstylePluginVersion=10.12.0
+spotbugsPluginVersion=5.0.14
+shadowJarPluginVersion=8.1.1
+downloadPluginVersion=5.4.0
+releasePluginVersion=2.8.0
+testngVersion=7.6.1
+eclipseLsp4jVersion=0.12.0
```
Comment (NipunaRanasinghe) [inline gradle.properties]: Why do we need testng and lsp4j dependencies for the excel connector?

File: ballerina/build.gradle [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    } else {
+        return extVersion.replace("${project.ext.snapshotVersion}", "")
+    }
+}
+
+ballerina {
+    packageOrganization = packageOrg
+    module = packageName
+    testCoverageParam = "--code-coverage --coverage-format=xml"
+    buildOnDockerImage = "nightly"
```
Comment (RDPerera) [inline ballerina/build.gradle]:
```suggestion
    isConnector = true
```

with new improvement to the Ballerina Gradle plugin

File: gradle.properties [hunk 5]
```diff
@@ -0,0 +1,12 @@
+org.gradle.caching=true
+group=io.ballerina.stdlib
+version=3.0.0-SNAPSHOT
+
```
Comment (RDPerera) [inline gradle.properties:5]:
```suggestion
ballerinaLangVersion=2201.8.0

```

File: gradle.properties [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+checkstylePluginVersion=10.12.0
+spotbugsPluginVersion=5.0.14
+shadowJarPluginVersion=8.1.1
+downloadPluginVersion=5.4.0
+releasePluginVersion=2.8.0
+testngVersion=7.6.1
+eclipseLsp4jVersion=0.12.0
+ballerinaGradlePluginVersion=2.1.8
```
Comment (RDPerera) [inline gradle.properties]:
```suggestion
ballerinaGradlePluginVersion=2.2.0
```
with improvement to the Ballerina Gradle plugin

File: build-config/resources/Ballerina.toml [hunk 7]
```diff
@@ -0,0 +1,17 @@
+[package]
+distribution = "2201.8.0"
+org = "ballerinax"
+name = "microsoft.excel"
+version = "@toml.version@"
+authors = ["Ballerina"]
+export = ["microsoft.excel"]
+repository = "https://github.com/ballerina-platform/module-ballerinax-googleapis.gmail"
```
Comment (RDPerera) [inline build-config/resources/Ballerina.toml]:
```suggestion
repository = "https://github.com/ballerina-platform/module-ballerinax-microsoft.excel"
```

File: .github/pull_request_template.md [hunk 8]
```diff
@@ -0,0 +1,12 @@
+## Purpose
```
Comment (RDPerera) [inline .github/pull_request_template.md:1]: Are we removing `issue_template.md` intentionally ?

**Review by RDPerera — CHANGES_REQUESTED** [review]

File: .github/pull_request_template.md (hunk 8 above)
Comment (kalaiyarasiganeshalingam) [reply .github/pull_request_template.md:1]: Yes, we will not create the issue in this repo. Hence, we don't need it.

File: changelog.md [hunk 9]
```diff
@@ -0,0 +1,11 @@
+# Changelog
+This file contains all the notable changes done to the Ballerina SQL package through the releases.
```
Comment (RDPerera) [inline changelog.md:2]: Should it be `Microsoft Excel` instead of `SQL`?

**Review by RDPerera — CHANGES_REQUESTED** [review]

**Comment by NipunaRanasinghe** [comment]
Closing this PR as the connector has been deprioritized due to shifting priorities. We will revisit and reopen it once it becomes a priority again.

**Pull request closed without merging**